File: hw/Vaca/u3/HW30ObserverAndStrategy/HW30Strategy/src/main/java/ec/edu/espe/controller/BubbleSort.java

package ec.edu.espe.controller;

/**
 *
 * @author MSI-PULSE
 */
public class BubbleSort implements SortingStrategy {

    @Override
    public void sort(int[] numbers) {
        int n = numbers.length;
        for (int i = 0; i < n-1; i++) {
            for (int j = 0; j < n-i-1; j++) {
                if (numbers[j] > numbers[j+1]) {
                    int temp = numbers[j];
                    numbers[j] = numbers[j+1];
                    numbers[j+1] = temp;
                }
            }
        }
    }

    @Override
    public String getStrategyName() {
        return "Bubble Sort";
    }
}

File: exams/gavilanez/u3/ExamP3(2)/ec_edu_espe_examp3_model/BubbleSort.py
from ec_edu_espe_examp3_controller.SortingStrategy import SortingStrategy

class BubbleSort(SortingStrategy):
    def sort(self, numbers):
        sorted_numbers = numbers[:]
        n = len(sorted_numbers)
        for i in range(n):
            for j in range(0, n-i-1):
                if sorted_numbers[j] > sorted_numbers[j+1]:
                    sorted_numbers[j], sorted_numbers[j+1] = sorted_numbers[j+1], sorted_numbers[j]
        return sorted_numbers